tests/integration/cmd_link_unlink_test.go

```diff
@@ -38,109 +37,59 @@ var _ = Describe("odo link and unlink command tests", func() {
 		})
 	})
 
-	/*
-		Context("When link between components using wrong port", func() {
-			JustBeforeEach(func() {
-				frontendContext = helper.CreateNewContext()
-				backendContext = helper.CreateNewContext()
-			})
-			JustAfterEach(func() {
-				helper.DeleteDir(frontendContext)
-				helper.DeleteDir(backendContext)
-			})
-			It("should fail", func() {
-				helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
-				helper.Cmd("odo", "create", "--s2i", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
-				helper.CopyExample(filepath.Join("source", "python"), backendContext)
-				helper.Cmd("odo", "create", "--s2i", "python", "backend", "--context", backendContext, "--project", commonVar.Project).ShouldPass()
-				helper.Cmd("odo", "push", "--context", backendContext).ShouldPass()
-				stdErr := helper.Cmd("odo", "link", "backend", "--context", frontendContext, "--port", "1234").ShouldFail().Err()
-				Expect(stdErr).To(ContainSubstring("Unable to properly link to component backend using port 1234"))
-			})
+	Context("When handling link/unlink between components", func() {
+		JustBeforeEach(func() {
+			frontendContext = helper.CreateNewContext()
+			backendContext = helper.CreateNewContext()
+		})
+		JustAfterEach(func() {
+			helper.DeleteDir(frontendContext)
+			helper.DeleteDir(backendContext)
+		})
+		It("should link the frontend application to the backend and then unlink successfully", func() {
+			helper.CopyExample(filepath.Join("source", "nodejs"), frontendContext)
+			helper.Cmd("odo", "create", "nodejs", "frontend", "--context", frontendContext, "--project", commonVar.Project).ShouldPass()
+			helper.Cmd("odo", "url", "create", "--port", "8080", "--context", frontendContext).ShouldPass()
+			helper.Cmd("odo", "push", "--context", frontendContext).ShouldPass()
+			frontendURL := helper.DetermineRouteURL(frontendContext)
+			oc.ImportJavaIS(commonVar.Project)
+			helper.CopyExample(filepath.Join("source", "openjdk"), backendContext)
+			helper.Cmd("odo", "create", "java:8", "backend", "--project", commonVar.Project, "--context", backendContext).ShouldPass()
```

Comment:
We get this error when trying to create the component with `java:8`:

```
Devfile component type java:8 is not supported, please run `odo catalog list components` for a list of supported devfile component types
```

Reply:
That must be because I removed `--s2i` from the command. I'll fix that.